src/careamics/careamist.py

```diff
@@ -646,16 +637,42 @@ def predict(
                 # record datamodule
                 self.pred_datamodule = datamodule
 
-                return self.trainer.predict(
-                    model=self.model, datamodule=datamodule, ckpt_path=checkpoint
-                )
-
             else:
                 raise ValueError(
                     f"Invalid input. Expected a CAREamicsPredData instance, paths or "
                     f"NDArray (got {type(source)})."
                 )
 
+        # TODO: tidy up: move to a seperate function
+        predictions = self.trainer.predict(
+            model=self.model, datamodule=self.pred_datamodule, ckpt_path=checkpoint
+        )
+        # not sure if necessary
+        if len(predictions) == 0:
+            return predictions
+        is_tiled = tile_size is not None
```

Comment:
Can't we get tiling param from the datamodule ?

Reply:
Yeah tbh, I did this pretty quick as proof of concept! Still a draft, I will fix up after I have tackled some other bugs